bootloader/src/pyi_utils.c

```diff
@@ -94,6 +94,33 @@ static int argc_pyi = 0;
 static void process_apple_events_new();
 #endif
 
+char *
+pyi_strjoin(const char *first, const char *sep, const char *second){
+    /* join first and second string, using sep as separator.
+     * any of them may be either a zero-terminated string or NULL.
+     * sep will be only used if first and second string are not empty.
+     */
+    int first_len=0, sep_len=0, second_len=0;
+    char *result;
+    if(first)
```

Comment:
Please add a space in front of the parent and put the body into `{…;}`. This is how it is done in the remaining code.